es-core/src/GuiComponent.cpp

```diff
@@ -105,6 +105,26 @@ void GuiComponent::setSize(float w, float h)
     onSizeChanged();
 }
 
+float GuiComponent::getZIndex() const
+{
+	return zIndex;
+}
+
+void GuiComponent::setZIndex(float z)
+{
+	zIndex = z;
+}
+
+float GuiComponent::getDefaultZIndex() const
+{
+	return zIndex;
+}
+
+void GuiComponent::setDefaultZIndex(float z)
+{
+	zIndex = z;
+}
+
```

Comment:
Themes are not required to set z-index values and each element has an appropriate default value. 
 The separate getters/setters for value and default are necessary in order to restore the default value when switching themes.